icu4c/source/configure

```diff
@@ -9121,6 +9121,24 @@ $as_echo "$as_me: WARNING: unrecognized options: $ac_unrecognized_opts" >&2;}
 fi
 
 
+if test -z "$PYTHON3";
+then
+    echo "" > data/rules.mk
+else
+    echo "Spawning Python to generate data/rules.mk..."
```

Comment:
A few notes:

- You don't need to re-run Python if you make a change to an existing txt file.  The Makefile will take care of that for you.  You only need to re-run Python if you add or remove a data file.
- Running configure is pretty quick.  It usually takes only 10-15 seconds on my workstation.
- Lazy-initializing Makefile rules seems fragile; are you suggesting to lazy-init the `rules.mk` file and then include it afterward?  How do you resolve the order?
- On Windows, Python is executed at build time.  I left comments for why there is a difference.  Ultimately, the Makefile approach is better for Unix systems, because it allows massively parallel builds and automatically takes care of timestamp resolution.

Of course, the other option would be to implement "smart" timestamp resolution and parallel builds in Python.  However, this is not a small task and screams "fragile" to me.  I don't want to build a whole miniature build system out of this Python script, something many other projects have spent years perfecting.  Let gnumake do what it is good at, and let Python do what it is good at.  If/when we move to Bazel/CMake, we will revisit this conversation.